### PR TITLE
Refactor/generalflux

### PR DESCRIPTION
**Pull request by Morriden**
Refactored pages into FLUX.
Will delete old files once approved by upper management.

### CODE REVIEW

File: src/components/markdown/View.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.Document {
```
Comment (mehtaphysical) [inline src/components/markdown/View.css:1]: This should now be `.View`

File: src/containers/Document.js [hunk 2]
```diff
@@ -1,26 +1,28 @@
-import React, { PureComponent } from 'react';
-import Preview from '../components/markdown/Preview';
-import Editor from '../components/markdown/Editor';
-import styles from './Document.css';
+// import React, { PureComponent } from 'react';
```
Comment (mehtaphysical) [inline src/containers/Document.js:1]: delete this file if you dont want it

File: src/hooks/appContext.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  const { state } = useContext(AppContext);
+  return state;
+};
+
+export const useDispatch = () => {
+  const { dispatch } = useContext(AppContext);
+  return dispatch;
+};
+
+export const useMarkdown = () => {
```
Comment (mehtaphysical) [inline src/hooks/appContext.js:15]: use a selector to get your state

File: src/hooks/appContext.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React, { useContext } from 'react';
+
+export const AppContext = React.createContext();
+
+export const useState = () => {
```
Comment (mehtaphysical) [inline src/hooks/appContext.js:5]: `useSelector`

File: src/reducers/appReducer.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { UPDATE_MARKDOWN } from '../utils/actions';
+
+export const initialState = 
+  {
```
Comment (mehtaphysical) [inline src/reducers/appReducer.js:4]: start objects on the same line as `=`

File: src/components/markdown/View.jsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import Editor from './Editor';
+import Preview from './Preview';
+import { updateMarkdown } from '../../utils/actions';
+import styles from './View.css';
+
+const View = () => {
+  const markdown = useMarkdown();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
```
Comment (mehtaphysical) [inline src/components/markdown/View.jsx:12]: i don't think you need this use effect

**Comment by Morriden** [comment]
Updated all requests.

File: src/components/markdown/Preview.js [hunk 7]
```diff
@@ -1,15 +1,12 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import marked from 'marked';
 import styles from 'github-markdown-css/github-markdown.css';
+import { useSelector, getMarkdownState } from '../../hooks/appContext';
 
-const Preview = ({ markdown }) => {
+const Preview = () => {
+  const markdown = useSelector(getMarkdownState);
   const __html = marked(markdown);
```
Comment (mehtaphysical) [inline src/components/markdown/Preview.js:8]: add the ability to convert your markdown to a preview to your selectors. `getMarkdownPreview`

File: src/hooks/appContext.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  const { state } = useContext(AppContext);
+  return selectorFn(state);
+};
+
+export const useDispatch = () => {
+  const { dispatch } = useContext(AppContext);
+  return dispatch;
+};
+
+export const getMarkdownState = state => state.markdown;
```
Comment (mehtaphysical) [inline src/hooks/appContext.js:15]: call thing `getMarkdown` and put it in a selectors folder. Make sure to add a test

File: src/utils/actions.js [hunk 9]
```diff
@@ -0,0 +1,5 @@
+export const UPDATE_MARKDOWN = 'UPDATE_MARKDOWN';
```
Comment (mehtaphysical) [inline src/utils/actions.js:1]: put this in an actions directory instead of utils

**Comment by Morriden** [comment]
Thank you we will get on those.

**Pull request merged**